src/Components/AddToCartButton/AddToCartButton.jsx

```diff
@@ -0,0 +1,32 @@
+import styles from "./AddToCartButton.module.css";
+import { useProduct, useCart, useControl } from "../../hooks";
+import { ToastMsg } from "../";
+
+export const AddToCartButton = ({ productId }) => {
+  const { isLoading, addToCart, isAlreadyInCart } = useCart();
+  const { isInStock } = useProduct();
+  const { changeRouteOnClick } = useControl();
+
+  return (
+    <>
+      {isAlreadyInCart(productId) ? (
+        <button
+          className={`btn font-md btn-secondary ${styles.btnMd}`}
+          onClick={() => changeRouteOnClick("cart")}
+        >
+          Go to cart
+        </button>
+      ) : isInStock(productId) ? (
+        <button

```

Comment:
On click disable button till server call is complete.

Reply:
Done.